### ISSUE TITLE
Mains water temperature can go below freezing

### ISSUE THREAD

**Issue by jmaguire1** [NREL/OpenStudio-HPXML#1581]
I'm pretty confident that at one point we were doing this but lost it. The mains temperature should never go below freezing, even if the algorithm we use would suggest a lower mains temperature. We should just set the minimum temperature to be freezing here, should be relatively straightforward to implement but it's not there right now.

We currently provide a schedule for the daily mains temperature like this:
![image](https://github.com/NREL/OpenStudio-HPXML/assets/8376057/ebc13ecb-3d7d-479c-bfd0-aac111fc89a4)
This example is Fairbanks, AK and clearly goes below 0.

**Comment by shorowit** [comment]
Huh. I see that we added this check to BEopt's code, but that must have been after we had already ported the routine over to OpenStudio. I don't see any issues/commits/code related to this since we moved to OpenStudio. Anyway, good to have this fixed up.

**Comment by jmaguire1** [comment]
Yeah, I'm very sure we had this check (I think we added it around the time we noticed that it wouldn't work for the southern hemisphere) in BEopt, and somehow it never made it's way into OS-HPXML. I was surprised to see I could get a negative mains temperature, although I've only seen that happen in Alaska which might be how this escaped notice for so long. Easy to fix at least, glad we had a reason to check this!
